src/Workspaces/Core/Portable/Classification/AbstractClassificationService.cs

```diff
@@ -95,20 +98,6 @@ internal abstract class AbstractClassificationService : IClassificationService
             }
         }
 
-        private static bool IsFullyLoaded(Document document, CancellationToken cancellationToken)
-        {
-            var workspaceStatusService = document.Project.Solution.Services.GetRequiredService<IWorkspaceStatusService>();
-
-            // Importantly, we do not await/wait on the fullyLoadedStateTask.  We do not want to ever be waiting on work
-            // that may end up touching the UI thread (As we can deadlock if GetTagsSynchronous waits on us).  Instead,
-            // we only check if the Task is completed.  Prior to that we will assume we are still loading.  Once this
-            // task is completed, we know that the WaitUntilFullyLoadedAsync call will have actually finished and we're
-            // fully loaded.
-            var isFullyLoadedTask = workspaceStatusService.IsFullyLoadedAsync(cancellationToken);
-            var isFullyLoaded = isFullyLoadedTask.IsCompleted && isFullyLoadedTask.GetAwaiter().GetResult();
```

Comment:
this was the wonky bit.  we're fine now, as the only place that does blocking in tagging is in the copy/paste tagger, and it uses JTF like so:

```c#
                    _owner._threadingContext.JoinableTaskFactory.Run(async () =>
                    {
                        var snapshotSpan = new DocumentSnapshotSpan(document, spanToTag);

                        // When copying/pasting, ensure we have classifications fully computed for the requested spans
                        // for both semantic classifications and embedded lang classifications.
                        await ProduceTagsAsync(context, snapshotSpan, classificationService, options, ClassificationType.Semantic, cancellationToken).ConfigureAwait(false);
                        await ProduceTagsAsync(context, snapshotSpan, classificationService, options, ClassificationType.EmbeddedLanguage, cancellationToken).ConfigureAwait(false);
                    });
```